src/debug/ee/controller.cpp

```diff
@@ -7938,15 +7894,24 @@ void DebuggerJMCStepper::TriggerMethodEnter(Thread * thread,
     if (!fIsUserCode)
         return;
 
+    if ((m_eMode != cStepIn) && (m_fp != LEAF_MOST_FRAME))
+    {
+        // We can run a stack trace in managed code (eg, not a stub)
+        ControllerStackInfo info;
+        StackTraceTicket ticket(ip);
+        info.GetStackInfo(ticket, thread, LEAF_MOST_FRAME, NULL);
```

Comment:
This gets the debugger's view of the frame pointer which allows us to compare framepointers properly.